src/app/app.component.ts

```diff
@@ -17,11 +22,55 @@ export class AppComponent {
   userName: string;
   userImageDoc: HalDoc;
 
+  podcastStoreSub: Subscription;
+  podcasts: PodcastModel[];
+  filterStoreSub: Subscription;
+  filter: FilterModel;
+
   constructor(
     private auth: AuthService,
-    private cms: CmsService
+    private cms: CmsService,
+    public store: Store<any>
   ) {
-    auth.token.subscribe(token => this.loadAccount(token));
+    auth.token.subscribe(token => {
+      this.loadAccount(token);
+      if (token) {
+        cms.auth.subscribe((cmsAuth) => this.loadCmsSeries(cmsAuth));
+      }
+    });
+  }
+
+  ngOnInit() {
+    this.podcastStoreSub = this.store.select('podcast').subscribe((state: PodcastModel[]) => {
+      this.podcasts = state;
+
+      if (this.podcasts.length > 0) {
+        let selectedPodcast;
+        if (Env.CASTLE_TEST_PODCAST) {
+          selectedPodcast = this.podcasts.find((p: PodcastModel) => {
+            return p.feederId === Env.CASTLE_TEST_PODCAST.toString();
+          });
+        } else {
+          selectedPodcast = this.podcasts[0];
+        }
+        if (selectedPodcast &&
+          (!this.filter || (this.filter && this.filter.podcast && selectedPodcast.seriesId !== this.filter.podcast.seriesId))) {
+          this.store.dispatch(castleFilter({podcast: selectedPodcast}));
+        }
+      }
+    });
+
+    this.filterStoreSub = this.store.select('filter').subscribe((state: FilterModel) => {
+      if (state.podcast && (!this.filter || state.podcast.seriesId !== this.filter.podcast.seriesId)) {
+        this.filter = state;
+        this.getEpisodes(this.filter.podcast);
+      }
+    });
+  }
+
+  ngOnDestroy() {
```

Comment:
"do we need to unsubscribe from stores onDestroy?" yup, fixed

Reply:
👍